### DISCUSSION TITLE
The DataFrame class

### DISCUSSION THREAD

**Discussion by steinnymir** [OpenCOMPES/sed#83] [Ideas]
We discussed the idea of defining a specialized DataFrame class to handle single-events. Here is a list of functionalities we thought would be good to have for it:

- [ ] translation tables, aka alias for the column names using internal names
- [ ] physical units for columns
- [ ] load from multiple sources: list of files, directory, run numbers etc...
- [ ] dummy loader for testing and example
- [ ] save/export dataframe as parquet, hdf5, maybe even nexus
- [ ] incorporate static (meta)data in the dataframe properties/attributes
- [ ] add ability to exchange "slow" data between columns in the dataframe and static metadata

If there are further ideas/suggestions, this is a good place to voice them.

**Comment by zain-sohail** [comment]
![Module structure UML  - DataFrame](https://user-images.githubusercontent.com/10053031/203336798-8212c1a1-d212-4e5b-9b09-6f8b8066a4a4.png)

Roughly the structure we agreed upon. The methods in Dataframe class are temporary.

In case of doubt: The Dataframe class exists to provide methods to create the desired dataframes for sed. The source class exists as an abstract class so that the user implements the methods that can be called by processor. Depending on the config, processor then dynamically instantiates the appropriate source class.
